packages/mongo-nav/src/user-menu/UserMenu.tsx

```diff
@@ -154,19 +164,74 @@ interface UserMenuProps {
    */
   onProductChange?: React.MouseEventHandler;
 
-  urls: Required<URLSInterface>;
+  /**
+   * Object that supplies URL overrides to UserMenu component.
+   * Shape: { userMenu:{ cloud: { userPreferences, organizations, invitations, mfa }, university: { videoPreferences }, support: { userPreferences }, account: { homepage } }}
+   */
+  urls: URLSInterface;
 
-  hosts: Required<HostsInterface>;
+  /**
+   * Object that supplies host overrides to UserMenu component.
+   * Shape: { cloud, realm, charts, account, university, support }
+   * Defaults to the production homepages of each product
+   */
+  hosts: HostsInterface;
 }
 
+/**
+ * # UserMenu
+ *
+ * UserMenu component
+ *
+ * ```
+<UserMenu
+  account={account}
+  activeProduct="cloud"
+  onLogout={onLogout}
+  onProductChange={onProductChange}
+  urls={urls}
+  hosts={hosts}
+/>
+```
+ * @param props.account Object that contains information about the active user. {firstName: 'string', lastName: 'string', email: 'string'}
+ * @param props.activeProduct MongoDB product that is currently active: ['cloud', 'university', 'support'].
+ * @param props.onLogout Callback invoked after the user clicks log out.
+ * @param props.onProductChange Callback invoked after the user clicks a product.
+ * @param props.hosts Object where keys are MDB products and values are the desired hostURL override for that product, to enable `<UserMenu />` to work across all environments.
+ * @param props.urls Object to enable custom overrides for every `href` used in `<UserMenu />`.
+ */
 function UserMenu({
   account: { firstName, lastName, email, openInvitations },
   activeProduct,
   onLogout = () => {},
   onProductChange = () => {},
-  urls,
-  hosts,
+  urls: urlsProp,
+  hosts: hostsProp,
 }: UserMenuProps) {
+  const hosts = defaultsDeep(hostsProp, defaultHosts);
+
+  const defaultURLs = {
```

Comment:
Might wanna drop a comment that this is temporary.

Reply:
👍